src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -941,9 +941,11 @@ self =>
         val pos = (opPos union t.pos) makeTransparentIf rightAssoc
         val sel = atPos(pos)(Select(stripParens(t), op.encode))
         if (targs.isEmpty) sel
-        else {
-          /* if it's right-associative, `targs` are between `op` and `t` so make the pos transparent */
-          atPos((pos union targs.last.pos) makeTransparentIf rightAssoc) {
+        else { // if it's right-associative, `targs` are between `op` and `t` so make the pos transparent
+          val pos0 = pos.union(targs.last.pos)
+          val end0 = pos.source.indexWhere(_ == ']', targs.last.pos.end)
```

Comment:
or save that position `OpInfo`? Didn't test if this is correct:

```scala
      val targs = if (in.token == LBRACKET) exprTypeArgs() else Nil
      val opinfo = OpInfo(top, name, targs, offset, targEndOffset = in.offset) // <<<
```

Reply:
I did that and a bit more clean-up. Probably you didn't expect your comment to inspire such diligence.